apps/schema/src/components/TableView.vue

```diff
@@ -0,0 +1,230 @@
+          }}
+        </p>
+      </div>
+      <table class="table table-bordered">
+        <thead>
+          <tr class="hoverContainer">
+            <th style="width: 25%" scope="col">
+              Column
+              <IconAction
+                icon="plus"
+                @click="createColumn"
+                class="btn-sm hoverIcon"
+              />
+            </th>
+            <th style="width: 25%" scope="col">Definition</th>
+            <th scope="col">Description</th>
+          </tr>
+        </thead>
+        <Draggable v-model="table.columns" tag="tbody" @end="applyPosition">
+          <ColumnView
+            v-for="(column, columnIndex) in table.columns"
+            :key="
+              JSON.stringify(column) +
+              '_' +
+              columnIndex +
+              '_' +
+              table.columns.length
+            "
+            :style="
+              isSubclassDropped(column) ? 'text-decoration: line-through' : ''
+            "
+            v-model="table.columns[columnIndex]"
+            :schema="schema"
+            :schemaNames="schemaNames"
+            @input="$emit('input', table)"
+            @createColumn="createColumn"
+          />
+        </Draggable>
+      </table>
+    </div>
+  </div>
+</template>
+
+<style>
+.hoverIcon {
+  visibility: hidden;
+}
+
+.hoverContainer:hover .hoverIcon {
+  visibility: visible;
+}
+</style>
+
+<script>
+import { IconAction, IconDanger } from "molgenis-components";
+import columnTypes from "../columnTypes.js";
+import ColumnView from "./ColumnView.vue";
+import Draggable from "vuedraggable";
+import TableEditModal from "./TableEditModal.vue";
+
+export default {
+  components: {
+    TableEditModal,
+    IconAction,
+    ColumnView,
+    Draggable,
+    IconDanger,
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true,
+    },
+    schema: {
+      type: Object,
+      required: true,
+    },
+    schemaNames: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      table: {},
+      columnTypes,
+    };
+  },
+  methods: {
+    applyPosition() {
+      let position = 1;
+      this.table.columns.forEach((column) => (column.position = position++));
+      this.$emit("input", this.table);
+    },
+    validateName() {
+      if (!this.name) {
+        return "Table name is required";
+      }
+      if (this.schema.tables.filter((t) => t.name === this.name).length > 1) {
+        return "Table name must be unique within schema";
+      }
+    },
+    deleteTable(table) {
+      if (!table.drop) {
+        //need to do deep set otherwise vue doesn't see it
+        this.$set(table, "drop", true);
+      } else {
+        this.$set(table, "drop", false);
+      }
+    },
+    isSubclassDropped(column) {
+      if (column.table === this.table.name) {
+        return this.table.drop;
+      } else {
+        return this.table.subclasses.filter(
```

Comment:
This line causes console errors when editing/creating a table
suggestion: this.table.subclasses?.find(...)